workflow/controller/workflowpod.go

```diff
@@ -1005,7 +1005,8 @@ func (woc *wfOperationCtx) addArchiveLocation(tmpl *wfv1.Template) {
 		// User explicitly set the location. nothing else to do.
 		return
 	}
-	needLocation := woc.artifactRepository.IsArchiveLogs()
+	wfArchiveLogs := woc.execWf.Spec.ArchiveLogs != nil && *woc.execWf.Spec.ArchiveLogs
```

Comment:
This is a pointer to a bool, so it is is tri-valued. I'm assuming we should should also never log if this is false, but this logic does not seem to encapsulate that. 

I'd move all this login into `woc.IsArchiveLogs()`

Reply:
my thinking is:
if controller on: then must log
if controller false: then depending on wf and tmpl

tmpl has priority over wf:
if tmpl is defined use tmpl, otherwise wf

(the test case in this PR is base on this logic)